pyat/at/lattice/lattice_info.py

```diff
@@ -0,0 +1,100 @@
+from warnings import warn
+from math import pi
+import numpy
+from scipy.constants import physical_constants as cst
+import at       # for AtWarning, AtError
+from ..lattice import checktype, elements
+
+__all__ = ['get_energy', 'get_voltage', 'get_energy_loss']
+
+TWO_PI_ERROR = 1.E-4
+
+
+def get_energy(ring):
+    """
+    energy, periodicity = get_energy(ring) Get the ring energy
+    get_energy looks for the machine energy in:
+        1) the 1st 'RingParam' element
+        2) the 1st 'RFCavity' element
+
+    PARAMETERS
+        ring  Ring structure
+
+    OUPUT
+        energy          Ring energy in eV
+        periodicity     Number of periods to make 2pi bending
+    """
+    params = list(filter(checktype(elements.RingParam), ring))
+
+    if len(params) > 0:
+        energy = params[0].Energy
+        periodicity = params[0].Periodicity
+    else:
+        cavities = list(filter(checktype(elements.RFCavity), ring))
+        if len(cavities) > 0:
+            energy = cavities[0].Energy
+            dipoles = filter(checktype(elements.Dipole), ring)
+            theta = [elem.BendingAngle for elem in dipoles]
+            try:
+                nbp = 2.0 * pi / sum(theta)
+            except ZeroDivisionError:
+                warn(at.AtWarning('No bending in the cell, set "Periodicity" to 1'))
+                periodicity = 1
+            else:
+                periodicity = int(round(nbp))
+                if abs(periodicity - nbp) > TWO_PI_ERROR:
+                    warn(at.AtWarning('non-integer number of cells: {0} -> {1}'.format(nbp, periodicity)))
+        else:
+            raise at.AtError('Energy not defined (searched in "RingParam" and "Cavities")')
+
+    return energy, periodicity
+
+
+def get_voltage(ring):
+    """
+    voltage, harm_number = get_voltage(ring) Get the accelerating voltage
+
+    PARAMETERS
+        ring  Ring structure
+
+    OUPUT
+        voltage    Total RF voltage
+        harmnumber Harmonic number
+    """
+    cavities = list(filter(checktype(elements.RFCavity), ring))
```

Comment:
Some people would say that 

```
cavities = [elem for elem in ring if isinstance(elem, RFCavity)]
```
is more pythonic.

Reply:
True… Some utility functions in utils.py are probably useless.